File: Data Structures/js-data-structures/stacks.js
// Stack

class Stack{
    constructor(){
        this.items = [];
    }

    add(element){   // add an element at the top of the stack
        return this.items.push(element);
    }

    remove(){   // removes the the topmost element of the stack
        if(this.items.length === 0){
            return 'Undeflow'
        }

        return this.items.pop()
    }

    peek(){     // peek the topmost element of the stack
        return this.items[this.items.length - 1]
    }

    size(){     // return the size of the stack
        return this.items.length
    }

    print(){      // print stack
        let str = ""

        for(let i = 0; i < this.items.length; i++){
            str += this.items[i] + " ";
        }

        return str
    }
}

const stack = new Stack()

stack.add(100);
stack.add(200);
stack.add(300);
stack.add(400);
stack.remove();

console.log(stack.peek())
console.log(stack.size())
console.log(stack.print())

File: Data Structures/py-data-structures/stacks.py
# Stack

class Stack:
    def __init__(self):
        self.items = []

    def add(self, element):     # add an element at the top of the stack
        return self.items.append(element)
    
    def remove(self):   # remove the topmost element of the stack
        length = len(self.items)

        if length == 0:
            return "Underflow"
        
        return self.items.pop()
        
    def peek(self):     # peek the topmost element of the stack
        last = len(self.items) - 1
        return self.items[last]

    def size(self):     # return the size of the stack
        return len(self.items)
    
    def print(self):       # print stack
        string = ""

        for item in self.items:
            string += str(item) + " "
        
        return string

stack = Stack()

stack.add(10)
stack.add(20)
stack.add(30)
stack.add(40)
stack.remove()

print(stack.print())